### PR TITLE
Flash optimizations in loop() and status screen menu

### PR DESCRIPTION
**Pull request by gudnimg**
These are just small optimizations I have noticed. Thought I should commit them in case I see more or forget them.

The 2 commits save 22 bytes of flash. Removes one global variable

### CODE REVIEW

File: Firmware/Marlin_main.cpp [hunk 1]
```diff
@@ -9839,15 +9839,7 @@ if(0)
 						lcd_update_enable(true);
 */
 						eFilamentAction=FilamentAction::AutoLoad;
-						bFilamentFirstRun=false;
-						if(target_temperature[0]>=EXTRUDE_MINTEMP){
-							bFilamentPreheatState=true;
-//							mFilamentItem(target_temperature[0],target_temperature_bed);
-							menu_submenu(mFilamentItemForce);
```
Comment (gudnimg) [inline Firmware/Marlin_main.cpp]: I have not tested this line. Need to verify we actually need this call to `menu_submenu`

Comment (gudnimg) [reply Firmware/Marlin_main.cpp]: I dropped this commit

File: Firmware/ultralcd.cpp [hunk 2]
```diff
@@ -791,11 +792,9 @@ void lcd_status_screen()                          // NOT static due to using ins
 	if (lcd_status_update_delay)
 		lcd_status_update_delay--;
 	else
-		lcd_draw_update = 1;
```
Comment (leptun) [inline Firmware/ultralcd.cpp:794]: I'm not so sure this change is needed. Before a lcd_draw_update event from elsewhere would re-render the status screen instantly. Now, a pending lcd update has to wait for the lcd_status_update_delay to expire, which will surely introduce some annoying delays. I'd keep it here how it was before

Comment (gudnimg) [reply Firmware/ultralcd.cpp:794]: I see what you mean. Do you agree ca8d47a6da33e74b89a29b5c0a6bb1decac951bb is sufficient? I don't see a reason why `lcd_draw_update` needs to be assigned a value in `lcd_status_screen()`. But if it is set outside `lcd_status_screen()` then we can render the screen immediately. (Currently still saving 10 bytes overall :))

Comment (leptun) [reply Firmware/ultralcd.cpp:794]: Yeah, that should work

**Review by DRracer — APPROVED** [review]

**Pull request merged**